src/client-side-encryption/client_encryption.ts

```diff
@@ -653,7 +679,13 @@ export class ClientEncryption {
       socketOptions: autoSelectSocketOptions(this._client.options)
     });
 
-    const { v } = deserialize(await stateMachine.execute(this, context));
+    const timeoutContext = this._timeoutMS
+      ? new CSOTTimeoutContext({
+          timeoutMS: this._timeoutMS,
+          serverSelectionTimeoutMS: this._client.options.serverSelectionTimeoutMS
+        })
+      : undefined;
```

Comment:
If we do this, there's no need to resolve `timeoutMS` from the client in the constructor - because we'll obtain it from the client when we resolve options when instantiating contexts.